databases/week3/Meal_application.sql

```diff
@@ -0,0 +1,372 @@
+`when` datetime timestamp,
+  
+`max_reservations` int (10) unsigned NOT NULL,
+  
+`price` decimal (4) NOT NULL,
+  
+`created_date` date
+ NOT NULL
+);
+
+
+
+CREATE TABLE `Reservation` (
+  
+`id` int (10) unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY,
+  
+`number_of_guests` int (10) unsigned NOT NULL,
+  
+`meal_id` int (10) unsigned NOT NULL,
+  
+`created_date` date
+ NOT NULL);
+
+
+
+CREATE TABLE `Review` (
+  
+`id` int (10) unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY,
+  
+`title` varchar(255) NOT NULL,
+  
+`description` text NOT NULL,
+  
+`meal_id` int(10) unsigned  NOT NULL,
+  
+`stars` int (5) unsigned  NULL,
+  
+`created_date` date
+ NOT NULL,
+);
+
+
+
+INSERT INTO Meal (Title, `Description`, Location, `When`, Max_reservations, Price, Created_date) 
+VALUES ('Meal001', 
+'With tomat', 
+'København H', 
+'2021-01-10 14:30:00', 
+'6',
+29, 
+'2020-01-08 00:00:00'
+); 
+INSERT INTO Meal (Title, `Description`, Location, `When`, Max_reservations, Price, Created_date) 
+VALUES ('Meal002', 
+'without tomat', 
+'København H', 
+'2021-01-10 14:30:00', 
+'6',
+29, 
+'2020-01-08 00:00:00'
+); 
+
+INSERT INTO Reservation (Number_of_guests, Meal_id, Created_date) 
+VALUES (
+"2",
+"2", 
+'2020-01-11'
+);
+INSERT INTO Reservation (Number_of_guests, Meal_id, Created_date) 
+VALUES (
+"5",
+"1", 
+'2020-01-10'
+);
+
+INSERT INTO Review (Title, `Description`, Meal_id, Stars, Created_date) 
+VALUES (
+'Excellent', 
+'Excellent food',
+'2', 
+4, 
+'2020-02-14'
+);INSERT INTO Review (Title, `Description`, Meal_id, Stars, Created_date) 
+VALUES (
+'Very good', 
+'Very good food',
+'1', 
+3, 
+'2020-02-14'
+);
+
+-- Get meals that has a price smaller than a specific price fx 90
+
+select * from meal where price < 90
+
+
+-- Get meals that still has available reservations...
+
+select Meal.Id,Meal.Title, Meal.max_reservation ,sum(Reservation.number_of_guests) as total
+from Meal
+inner join Reservation on Reservation.meal_id = Meal.Id
+having total < max_reservation;
+
+-- Get meals that partially match a title. Rød grød med will match the meal with the title Rød grød med fløde...
+
+SELECT * from meal   
+where Meal.Title like '%Rød grød med%';
+
+--Get meals that has been created between two dates...
+SELECT * FROM meal 
+WHERE created_date BETWEEN '2020-01-10' AND '2020-01-11';
+
+-- Get only specific number of meals fx return only 5 meals
+select * from meal
+limit 5;
+
+-- Get the meals that have good reviews...
+select meal.*, review.title, review.stars
+from meal 
+inner join review on review.review_meal_id = meal.id
+where review.stars > 3;
+
+-- Get reservations for a specific meal sorted by created_date...
+select meal.*, reservation.created_date 
+from reservation
+inner join meal on meal.id = reservation.meal_id
+where meal.id = 2
+ORDER by reservation.created_date ASC;
+
+-- Sort all meals by average number of stars in the reviews
+select meal.title, AVG(review.stars) from meal 
+inner join review on review.review_meal_id = meal.id

```

Comment:
please change this `review.review_meal_id = meal.id `to `review.meal_id = meal.id`